shared/util/container.tsx

```diff
@@ -19,14 +20,28 @@ export const networkErrorCodes = [
 
 export const isNetworkErr = (code: number) => networkErrorCodes.includes(code)
 
-export const getRouteProps = (ownProps: any, key: string) => ownProps.navigation.getParam(key)
+export function getRouteProps <T = any>(ownProps: {navigation: {getParam: (key: string) => unknown}}, key: string) { return ownProps.navigation.getParam(key) as T }
 
 export type TypedDispatch = (action: TypedActions) => void
 export type Dispatch = TypedDispatch
 
 export const useAnyWaiting = (...waitingKeys: string[]) =>
   useSelector(state => anyWaiting(state, ...waitingKeys))
 export const useAnyErrors = (...waitingKeys: string[]) => useSelector(state => anyErrors(state, waitingKeys))
+export function usePrevious<T>(value: T) {
+  const ref = React.useRef<T>()
+  React.useEffect(() => {
+    ref.current = value
+  })
+  return ref.current
```

Comment:
This is kinda tricky, but interesting that it works. Here's my understanding:

You create a closure that contains the `value`. You pass that closure to useEffect so that after a render it sets the mutable `.current` field of the `ref` to the value. 

The function returns ref.current which will refer to the value set after the previous render.

Reply:
Maybe a comment like the above, should be included as usage doc, because it may not be straightforward to use.

Reply:
see https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state